KIPs/kip-17-non_fungible_token.md

```diff
@@ -0,0 +1,528 @@
+    ///  (sort order not specified)
+    function tokenByIndex(uint256 _index) external view returns (uint256);
+
+    /// @notice Enumerate NFTs assigned to an owner
+    /// @dev Throws if `_index` >= `balanceOf(_owner)` or if
+    ///  `_owner` is the zero address, representing invalid NFTs.
+    /// @param _owner An address where we are interested in NFTs owned by them
+    /// @param _index A counter less than `balanceOf(_owner)`
+    /// @return The token identifier for the `_index`th NFT assigned to `_owner`,
+    ///   (sort order not specified)
+    function tokenOfOwnerByIndex(address _owner, uint256 _index) external view returns (uint256);
+}
+```
+
+### Minting Extension
+The **minting extension** is OPTIONAL for KIP-17 smart contracts. This allows your contract to mint a new token.
+
+```solidity
+pragma solidity 0.4.24;
+
+/// @title KIP-17 Non-Fungible Token Standard, optional minting extension
+///  Note: the KIP-13 identifier for this interface is 0xeab83e20.
+interface KIP17Mintable {
+    /// @notice Create a new token
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    /// @param _to The account that will receive the minted token
+    /// @param _tokenId The token ID to mint
+    /// @return True if the minting operation is successful, false otherwise
+    function mint(address _to, uint256 _tokenId) public returns (bool);
+
+    /// @notice Check the minting permission
+    /// @param _account The account to check the minting permission
+    /// @return True if the account has the minting permission, false otherwise
+    function isMinter(address _account) public view returns (bool);
+
+    /// @notice Give the minting permission to `_account`
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    /// @param _account The account to be given the minting permission
+    function addMinter(address _account) public;
+
+    /// @notice Renounce the minter permission of `msg.sender`
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    function renounceMinter() public;
+}
+```
+
+### Minting with URI Extension
+The **minting with URI extension** is OPTIONAL for KIP-17 smart contracts. This allows your contract to mint a new token with URI.
+
+```solidity
+pragma solidity 0.4.24;
+
+/// @title KIP-17 Non-Fungible Token Standard, optional minting with URI extension
+///  Note: the KIP-13 identifier for this interface is 0x50bb4e7f.
+interface KIP17MetadataMintable {
+    /// @notice Create a new token with the specified URI
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    /// @param _to The account that will receive the minted token
+    /// @param _tokenId The token ID to mint
+    /// @param _tokenURI the token URI of the newly minted token
+    /// @return True if the minting operation is succeeded, false otherwise
+    function mintWithTokenURI(address _to, uint256 _tokenId, string memory _tokenURI) public returns (bool);
```

Comment:
```suggestion
    function mintWithTokenURI(address _to, uint256 _tokenId, string memory _tokenURI) public returns (bool);
```
Implementer can choose either mintable or metadataMintable to implement one based on their needs.
Therefore, I think that the definition of addMinter / renounceMinter / isMinter functions should be defined here. What do you think?

Reply:
Done.